pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultiStageQueryThrottler.java

```diff
@@ -193,4 +204,24 @@ public int currentQueryServerThreads() {
   int availablePermits() {
     return _semaphore.availablePermits();
   }
+
+  @VisibleForTesting
+  int calculateMaxServerQueryThreads(PinotConfiguration config) {
+    Map<String, String> clusterConfigMap = _helixAdmin.getConfig(
+        _helixConfigScope,
+        Collections.singletonList(CommonConstants.Helix.CONFIG_OF_MULTI_STAGE_ENGINE_MAX_SERVER_QUERY_THREADS));
+    int maxServerQueryThreadsFromClusterConfig = Integer.parseInt(
+        clusterConfigMap.getOrDefault(
+            CommonConstants.Helix.CONFIG_OF_MULTI_STAGE_ENGINE_MAX_SERVER_QUERY_THREADS,
+            CommonConstants.Helix.DEFAULT_MULTI_STAGE_ENGINE_MAX_SERVER_QUERY_THREADS));
+    int maxServerQueryThreadsFromBrokerConfig = config.getProperty(
+            CommonConstants.Broker.CONFIG_OF_MSE_MAX_SERVER_QUERY_THREADS,
+            CommonConstants.Broker.DEFAULT_MSE_MAX_SERVER_QUERY_THREADS);
```

Comment:
Given this one doesn't change, we can calculate it in the constructor and store it in a member variable

Reply:
Since instance level override has higher priority, we can consider always use the instance level one when it exists. We calculate the cluster level one only if instance level one doesn't exist